applications/launcher/widgets/SettingsPopup.qml

```diff
@@ -49,28 +49,28 @@ Item {
                     stepSize: 1
                     value: controller.sleepAfter
                     onDownPressedChanged: {
-                        if(this.value <= 10){
-                            this.stepSize = 1;
-                            return;
-                        }
-                        if(this.value <= 60){
-                            this.stepSize = 5;
-                            return;
-                        }
-                        this.stepSize = 15;
+                        this.lastDirectionUp = false;
                     }
                     onUpPressedChanged: {
-                        if(this.value < 10){
-                            this.stepSize = 1;
+                        this.lastDirectionUp = true;
+                    }
+                    onValueChanged: {
+                        if(this.ignoreNextValueChange){
```

Comment:
Ideally this logic should be moved into BetterSpinBox, and it should only call the valueChanged signal if ignoreNextValueChange is false.

Reply:
It may be best to create a second signal. I'm thinking "valueStable"? It doesn't look like Javascript's `super` keyword will be any use here at all.

Unfortunately, all of this can be overwritten if someone intuitively attempts to use the original `onValueChanged` signal.

Reply:
Or, we could just try to make `ignoreNextValueChange` an instance variable and not declare it in BetterSpinBox at all.

Reply:
What you can do is make Item be the root of BetterSpinBox and then declare the signals/properties there and map them into the SpinBox. That way you'll have more control over when things are called.